exonum-java-binding-core/rust/src/storage/proof_map_index.rs

```diff
@@ -162,16 +173,92 @@ pub extern "system" fn Java_com_exonum_binding_storage_indices_ProofMapIndexProx
 ) -> jobject {
     let res = panic::catch_unwind(|| {
         let key = convert_to_key(&env, key)?;
-        env.ensure_local_capacity(512)?;
```

Comment:
You must either call this or delete-as-you-go. Just have to pick the right size. I'd use _2 * (num(branches) + num(leaves))_

Reply:
:thinking: But we don't know the count of branches and leaves at this moment. What about using `auto_local`?

Reply:
Yes, it will complicate thing. We can get the number of branches just below, so let's apply a simple estimate of _100 + 2 * num(branches)_ (you won't request a single proof for a hundred keys, will you?).